spec/models/dfe_signin/authorisation_spec.rb

```diff
@@ -0,0 +1,23 @@
+require "rails_helper"
+
+RSpec.describe DfeSignin::Authorisation do
+  subject { described_class.new(organisation_id: 123, user_id: 456) }
+
+  describe "authorised?" do
+    context "when user has access to the service" do
+      before { allow(subject).to receive(:roles).and_return(["Role 1", "Role 2"]) }
```

Comment:
Stubbing this internal method makes this spec brittle to changes to the implementation. It also means the code that parses the response body to pull out the roles does not have unit test coverage. Can these specs be re-written without any stubbing?